kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -365,6 +376,27 @@ static SchemaChanges computeSchemaChangesById(StructType currentSchema, StructTy
     while (newSchemaIterator.hasNext()) {
       SchemaIterable.MutableSchemaElement newElement = newSchemaIterator.next();
       SchemaElementId id = getSchemaElementId(newElement);
+
+      // If the element is a struct field we need to validate that it has not been moved out of
+      // its parent struct. To do this, we check that in the old schema and the new schema
+      // its parent struct field (and the path to it) is unchanged.
+      if (newElement.isStructField()) {
```

Comment:
I considered first detecting these sorts of moves, and then adding them as a different list of changes (maybe `fieldMoves`) to `SchemaDiff`, but considering the logic later would be to just throw if any of them existed I decided this was unnecessary. If we later have a need to do (1) detection and (2) throw in separate steps we can fix this then.